pages/seller/coupon/assign.js

```diff
@@ -0,0 +1,77 @@
+import CommerceLayout from '@components/common/CommerceLayout';
+import SiteHead from '@components/common/SiteHead.js';
+import { GET } from '@apis/defaultApi';
+import { useEffect, useState } from 'react';
+import CouponList from '@components/coupon/CouponList';
+import styled from '@emotion/styled';
+import UserSearchBar from '@components/coupon/UserSearchBar';
+
+export default function CouponAssign() {
+  const uid = 4;
+
+  const [role, setRole] = useState('');
+  const [userId, setUserId] = useState(uid);
+  const [sellerId, setSellerId] = useState();
+
+  // TODO:
+  // setUserId(현재로그인되어있는userId-세션,쿠키 등에서 얻어올 것임);
+
+  useEffect(() => {
+    GET(`/user/role`, { userId }).then((res) => {
+      if (res.message === 'ROLE_USER') {
+        setRole('USER');
+      }
+      if (res.message === 'ROLE_SELLER') {
+        setRole('SELLER');
+        setSellerId(res.data);
+      }
+    });
+  }, []);
+  const userProps = {
+    userId: userId,
+  };
+  const sellerProps = {
+    sellerId: sellerId,
+  };
+  if (role === 'USER') {
+    return (
+      <CommerceLayout>
+        <SiteHead title="User's Coupon List" />
+        <p>사용자는 쿠폰을 생성할 수 없습니다.</p>
+      </CommerceLayout>
+    );
+  }
+  if (role === 'SELLER') {
+    return (
+      <CommerceLayout>
+        <SiteHead title="Seller's Coupon List" />
+        <PageContainer>
+          <CouponSection>
+            <CouponList {...sellerProps}></CouponList>
+          </CouponSection>
+          <SearchbarSection>
+            <UserSearchBar></UserSearchBar>
+          </SearchbarSection>
+        </PageContainer>
+      </CommerceLayout>
+    );
+  }
```

Comment:
예외처리를 꼼꼼하게 하는 게 좋을 것 같아요

```suggestion
  if (role === 'USER') {
    return (
      <CommerceLayout>
        <SiteHead title="User's Coupon List" />
        <p>사용자는 쿠폰을 생성할 수 없습니다.</p>
      </CommerceLayout>
    );
  }
  else if (role === 'SELLER') {
    return (
      <CommerceLayout>
        <SiteHead title="Seller's Coupon List" />
        <PageContainer>
          <CouponSection>
            <CouponList {...sellerProps}></CouponList>
          </CouponSection>
          <SearchbarSection>
            <UserSearchBar></UserSearchBar>
          </SearchbarSection>
        </PageContainer>
      </CommerceLayout>
    );
  } else {
    return(<CommerceLayout>
        <SiteHead title="Coupon List" />
        <p>잘못된 접근입니다.</p>
      </CommerceLayout>);
  }
```